prow/plank/controller.go

```diff
@@ -266,24 +270,7 @@ func (c *Controller) SyncMetrics() {
 // in-place when it aborts.
 // TODO: Dry this out - need to ensure we can abstract children cancellation first.
 func (c *Controller) terminateDupes(pjs []prowapi.ProwJob, pm map[string]coreapi.Pod) error {
-	// "job org/repo#number" -> newest job
-	dupes := make(map[string]int)
-	for i, pj := range pjs {
-		if pj.Complete() || pj.Spec.Type != prowapi.PresubmitJob {
-			continue
-		}
-		n := fmt.Sprintf("%s %s/%s#%d", pj.Spec.Job, pj.Spec.Refs.Org, pj.Spec.Refs.Repo, pj.Spec.Refs.Pulls[0].Number)
-		prev, ok := dupes[n]
-		if !ok {
-			dupes[n] = i
-			continue
-		}
-		cancelIndex := i
-		if (&pjs[prev].Status.StartTime).Before(&pj.Status.StartTime) {
-			cancelIndex = prev
-			dupes[n] = i
-		}
-		toCancel := pjs[cancelIndex]
+	return c.aborter.TerminateOlderPresubmitJobs(pjs, func(toCancel prowapi.ProwJob) error {
```

Comment:
This never returns an error

Reply:
That's right but this is how it was originally implemented. It is just refactored but keeping the reconcile loop behaviour the same.